core/primitives/src/state_record.rs

```diff
@@ -28,3 +37,104 @@ pub enum StateRecord {
         data: Option<Vec<u8>>,
     },
 }
+
+impl StateRecord {
+    pub fn from_raw_key_value(key: Vec<u8>, value: Vec<u8>) -> Option<StateRecord> {
+        let column = &key[0..1];
+        match column {
+            col::ACCOUNT => {
+                let separator = (1..key.len()).find(|&x| key[x] == ACCOUNT_DATA_SEPARATOR[0]);
+                if separator.is_some() {
```

Comment:
I know that you just moved the code, but I still review what we have, and it seems that this is equivalent to

```rust
if key[1..].contains(ACCOUNT_DATA_SEPARATOR[0]) {
```

Reply:
+1